examples/demo_dlmbl/python/excercise_1.py

```diff
@@ -0,0 +1,281 @@
+    ...
+    # plot one image from each of the batch and break
+    break
+
+# %% tags=["solution"]
+train_dataloader = data_module.train_dataloader()
+
+
+fig, axs = plt.subplots(3, 8, figsize=(20, 6))
+
+# Draw 8 batches, each with 32 images. Show the first image in each batch.
+
+for i, batch in enumerate(train_dataloader):
+    # The batch is a dictionary consisting of three keys: 'index', 'source', 'target'.
+    # index is the tuple consisting of (image name, time, and z-slice)
+    # source is the tensor of size 1x1x256x256
+    # target is the tensor of size 2x1x256x256
+
+    if i >= 8:
+        break
+    FOV = batch["index"][0][0]
+    input_tensor = batch["source"][0, 0, :, :].squeeze()
+    target_membrane_tensor = batch["target"][0, 0, :, :].squeeze()
+    target_nuclei_tensor = batch["target"][0, 1, :, :].squeeze()
+
+    axs[0, i].imshow(input_tensor, cmap="gray")
+    axs[1, i].imshow(target_nuclei_tensor, cmap="gray")
+    axs[2, i].imshow(target_membrane_tensor, cmap="gray")
+    axs[0, i].set_title(f"input@{FOV}")
+    axs[1, i].set_title("target-nuclei")
+    axs[2, i].set_title("target-membrane")
+    axs[0, i].axis("off")
+    axs[1, i].axis("off")
+    axs[2, i].axis("off")
+
+plt.tight_layout()
+plt.show()
+
+
+# %% [markdown]
+"""
+Construct a 2D U-Net for image translation.
+
+See ``viscy.unet.networks.Unet2D.Unet2d`` for configuration details.
+Increase the ``depth`` in ``draw_graph`` to zoom in.
+"""
+
+# %%
+model_config = {
+    "architecture": "2D",
+    "in_channels": 1,
+    "out_channels": 2,
+    "residual": True,
+    "dropout": 0.1,
+    "task": "reg",
+}
+
+model = VSUNet(
+    model_config=model_config.copy(),
+    batch_size=BATCH_SIZE,
+    loss_function=torch.nn.functional.mse_loss,
+    schedule="WarmupCosine",
+    log_num_samples=10,
+)
+
+# visualize graph
+model_graph = draw_graph(model, model.example_input_array, depth=2, device="cpu")
+graph = model_graph.visual_graph
+graph
+
+# %% [markdown]
+"""
+Configure trainer class.
+Here we use the ``fast_dev_run`` flag to run a sanity check first.
+"""
+
+# %%
+trainer = VSTrainer(accelerator="gpu", device=GPU_ID, fast_dev_run=True)
+
+trainer.fit(model, datamodule=data_module)
+
+# %% [markdown]
+"""
+<div class="alert alert-info">
+Task 1.3
+
+Modify the trainer to train the model for 20 epochs.
+</div>
+"""
+
+# %% [markdown]
+"""
+Tips:
+
+- See ``VSTrainer?`` for all the available parameters.
+- Set ``default_root_dir`` to store the logs and checkpoints
+in a specific directory.
+"""
+
+# %% [markdown]
+"""
+Bonus:
+
+- Tweak model hyperparameters
+- Adjust batch size to fully utilize the VRAM
+"""
+
+# %% tags=["solution"]
+wider_config = model_config | {"num_filters": [24, 48, 96, 192, 384]}
+
+model = model = VSUNet(
+    model_config=wider_config.copy(),
+    batch_size=BATCH_SIZE,
+    loss_function=torch.nn.functional.mse_loss,
+    schedule="WarmupCosine",
+    log_num_samples=10,
+)
+
+trainer = VSTrainer(
```

Comment:
I think we should write the logs in pwd, such that the folder structure is 04-image-translation/setup.sh, 04-image-translation/viscy, 04-image-translation/lightning_logs, ...